### PR TITLE
feat: popover-menu component 

### PR DESCRIPTION
**Pull request by albinscania**
**Describe pull-request**  
A new component in the form of a Popover Menu, based on core-code from the popover canvas component.

**Solving issue**  
Fixes: #2043

**How to test**  
Go to the netlify link or checkout the branch, and browse to the PopoverMenu component. Click the kebab-dots to see it in action, both with and without icon.

Now with updated z-index to be correctly placed.

### CODE REVIEW

**Comment by alexlioliopoulos** [comment]
It is grey-100! 

(sdds-library in Figma is usually the source of truth when in doubt)
image

**Comment by alexlioliopoulos** [comment]
Looks great as Matylda said! 

It is a tradition though to have at least one question in every PR so I won't break it here :D 

It might not have been clear on the task itself, but the idea was that either have icons for all items, or for none of them. Could we set it to be like that? Also we want devs using this to be able to add a divider wherever they want. Is it flexible like that?

**Comment by albinscania** [comment]
@alexlioliopoulos Ah I see... I guess I can add something like an "with-icon" class and then just indent the text where they choose not to add an icon? (even if that's against the design).

And yes, dividers can be added anywhere :)

**Comment by alexlioliopoulos** [comment]
> 

That would be an even greater solution - let's go with it if it is easy to make!

File: components/src/components/popoverMenu/popovermenu.scss [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    &:last-child {
+      margin-bottom: 0;
+    }
+
+    a {
+      text-decoration: none;
+      position: relative;
+      width: 100%;
+      display: block;
```
Comment (theJohnnyMe) [inline components/src/components/popoverMenu/popovermenu.scss:38]: As one on 40 is flex, this one can be removed.

File: components/src/components/popoverMenu/popovermenu.scss [hunk 2]
```diff
@@ -0,0 +1,94 @@
+@import "../../helpers/style-helpers.scss";
+@import "@scania/typography/dist/scss/mixins";
+@import "@scania/typography/dist/scss/tokens";
+
+.sdds-popover-menu {
+  overflow: hidden;
+  width: 160px;
+  background-color: #fff;
```
Comment (theJohnnyMe) [inline components/src/components/popoverMenu/popovermenu.scss:8]: Not a biggy but could be var(--sdds-white)...

File: components/src/components/popoverMenu/popovermenu.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    this.popoverMenuElement.addEventListener('mousedown', (event) => {
+      event.stopPropagation();
+    });
+  }
+
+  render() {
+    return (
+      <div
+        ref={(el) => (this.popoverMenuElement = el as HTMLElement)}
```
Comment (theJohnnyMe) [inline components/src/components/popoverMenu/popovermenu.tsx:88]: As in popover canvas, this is a good chance to use Host :)

File: components/src/components/popoverMenu/popovermenu.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import { Component, Listen, h, Prop, State } from '@stencil/core';
+import { createPopper } from '@popperjs/core';
+import type { Placement } from '@popperjs/core';
+
+@Component({
+  tag: 'sdds-popover-menu',
+  styleUrl: 'popovermenu.scss',
```
Comment (theJohnnyMe) [inline components/src/components/popoverMenu/popovermenu.tsx:7]: as in popover canvas, see if folder name and files in can be renamed to popover-menu, popover-menu.scss etc... just to align with rest of "two-words" components...

File: components/src/components/popoverMenu/popovermenu.stories.js [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+          'right',
+          'right-start',
+          'right-end',
+        ],
+      },
+      defaultValue: 'right-start',
+      description: 'Position of the PopoverMenu',
+    },
+    withIcons: {
+      name: 'withIcons',
```
Comment (theJohnnyMe) [inline components/src/components/popoverMenu/popovermenu.stories.js:29]: In the version of the story, I checked there were no icons showing when this one was true. Please check.

**Review by theJohnnyMe — CHANGES_REQUESTED** [review]

**Review by theJohnnyMe — APPROVED** [review]
Looks good to me.

**Pull request merged**